src/metrics/peerstore.go

```diff
@@ -359,6 +361,41 @@ func (c *PeerStore) ExportToCSV(filePath string) error {
 	return nil
 }
 
+// Function that save the raw Data of all the peers to a db (Using BoltDB)
+// Using the peerID as a key, and the Peer struct marshalled into JSON as a value
+func (c *PeerStore) PersistToDb() {
+	db, err := bolt.Open("../../peerstore.db", 0600, &bolt.Options{Timeout: 5 * time.Second})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	err = db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists([]byte("PeerStore"))
+		if err != nil {
+			return err
+		}
+
+		c.PeerStore.Range(func(k, val interface{}) bool {
+			peerData := val.(Peer)
+			buf, err := json.Marshal(peerData)
+			if err != nil{
+				log.Fatal(err)
```

Comment:
Instead of logging Fatal, which stops the execution of the program lets return the error we got with some extra information.
As a tip, I think its good to use Fatal only under some very [specific circumstances](https://stackoverflow.com/questions/33885235/should-a-go-package-ever-use-log-fatal-and-when).